staging/src/k8s.io/dynamic-resource-allocation/structured/allocator.go

```diff
@@ -1094,68 +1118,110 @@ func taintTolerated(taint resourceapi.DeviceTaint, request requestAccessor) bool
 	return false
 }
 
-func (alloc *allocator) checkAvailableCapacity(device deviceWithID) (bool, error) {
+// checkAvailableCounters checks if there are enough counters available to allocate
+// the specified device.
+//
+// Gets called only if the partitionable devices feature is enabled and the device
+// consumes counters.
+func (alloc *allocator) checkAvailableCounters(device deviceWithID) (bool, error) {
 	slice := device.slice
-
-	referencedSharedCounters := sets.New[draapi.UniqueString]()
-	for _, consumedCounter := range device.basic.ConsumesCounters {
-		referencedSharedCounters.Insert(consumedCounter.CounterSet)
-	}
-
-	// Create a structure that captures the initial counter for all sharedCounters
-	// referenced by the device.
-	availableCounters := make(map[draapi.UniqueString]map[string]draapi.Counter)
-	for _, counterSet := range slice.Spec.SharedCounters {
-		if !referencedSharedCounters.Has(counterSet.Name) {
-			// the API validation logic has been added to make sure the counterSet referred should exist in capacityPools
-			continue
-		}
-		counterShared := make(map[string]draapi.Counter, len(counterSet.Counters))
-		for name, cap := range counterSet.Counters {
-			counterShared[name] = cap
+	sliceName := draapi.MakeUniqueString(slice.Name)
```

Comment:
Capturing some random thoughts:
- We could convert `slice.Name` already via `k8s.io/dynamic-resource-allocation/api` -  a bit involved, because it's in the ObjectMeta. Only worth if it also helps elsewhere?
- We convert here (current code).
- We don't convert and use normal string lookups in the maps. Has this been considered?

`MakeUniqueString` is not cheap, so it's not obvious which of the last two options is faster.

Reply:
I think for now the best solution here is to not convert, so I've updated the PR to do that. I want to have a more detailed discussion about when to use UniqueString in https://github.com/kubernetes/kubernetes/pull/131357 as it adds additional fields with strings. Also I'm having some trouble with the autoconversion and the UniqueString type.